packages/devtools/lib/src/ui/gtags.dart

```diff
@@ -0,0 +1,221 @@
+/// *** DevTools Property Specific QA events and dimensions.                 ***
+/// ****************************************************************************
+
+// Type of events (event_category):
+const String gaApplicationState = 'application_state'; // type, build, platform
+const String gaScreenViewEvent = 'screen'; // Active screen (tab selected).
+const String gaSelectEvent = 'select'; // User selected something.
+
+const String gaOnError = 'onerror'; // Browser onError detected in DevTools
+const String gaException = 'exception'; // Any Dart exception in DevTools
+
+// DevTools GA screenNames
+const String gaDevTools = 'main';
+const String gaDebugger = 'debugger';
+const String gaInspector = 'inspector';
+const String gaMemory = 'memory';
+const String gaTimeline = 'timeline';
+const String gaLogging = 'loggimng';
+
+// DevTools UI action selected (clicked).
+
+// Main bar UX actions:
+const String gaHotReload = 'hotReload';
+const String gaHotRestart = 'hotRestart';
+const String gaFeedback = 'feedback';
+
+// Common UX actions:
+const String gaPause = 'pause'; // Memory, Timeline, Debugger
+const String gaResume = 'resume'; // Memory, Timeline, Debugger
+
+// Inspector UX actions:
+const String gaWidgetMode = 'widgetMode';
+const String gaRefresh = 'refresh';
+const String gaPerformanceOverlay = 'overlay';
+const String gaDebugPaint = 'debugPaint';
+const String gaPaintBaseline = 'paintBaseline';
+const String gaSlowAnimation = 'slowAnimation';
+const String gaRepaintRainbow = 'repaintRainbow';
+const String gaDebugBanner = 'debugBanner';
+const String gaTrackRebuilds = 'rebuilds';
+const String gaIOS = 'iOS';
+const String gaSelectWidgeMode = 'selectWidgeMode';
+
+// Timeline UX actions:
+const String gaTimelineFrame = 'frame'; // Frame selected in frame chart
+const String gaTimelineFlame = 'flame'; // Select a UI/GPU flame
+
+// Memory UX actions:
+const String gaSnapshot = 'snapshot';
+const String gaReset = 'reset';
+const String gaGC = 'gc';
+const String gaInspectClass = 'inspectClass'; // inspect a class from snapshot
+const String gaInspectInstance = 'inspectInstance'; // inspect an instance
+const String gaInspectData = 'inspectData'; // inspect data of the instance
+
+// Debugger UX actions:
+const String gaOpenShortcut = 'openShortcut';
+const String gaStepIn = 'stepIn';
+const String gaStepOver = 'stepOver';
+const String gaStepOut = 'stepOut';
+const String gaBP = 'bp';
+const String gaUnhandledExceptions = 'unhandledExceptions';
+const String gaAllExceptions = 'allExceptions';
+
+// Logging UX actions:
+const String gaClearLogs = 'clearLogs';
+
+void gaScreen(String screenName, [int value = 0]) => GTag.event(
+    screenName,
+    GtagEvent(
+        event_category: gaScreenViewEvent,
+        value: value,
+        app_type: userAppType,
+        build_type: userBuildType,
+        platform_type: userPlatformType));
+
+void gaSelect(String screenName, String selectedItem, [int value = 0]) =>
+    GTag.event(
```

Comment:
I don't think dartfmt handles this case well.
I would use a trailing comma after the last line and use
```
{
}
```
instead of `=>`

Reply:
Yeah looks better. Done